credentials/credentials.go

```diff
@@ -190,16 +196,16 @@ func NewTLS(c *tls.Config) TransportCredentials {
 }
 
 // NewClientTLSFromCert constructs a TLS from the input certificate for client.
-// serverNameOverwrite is for testing only. If set to a non empty string,
-// it will overwrite the virtual host name of authority (e.g. :authority header field) in requests.
-func NewClientTLSFromCert(cp *x509.CertPool, serverNameOverwrite string) TransportCredentials {
-	return NewTLS(&tls.Config{ServerName: serverNameOverwrite, RootCAs: cp})
+// serverNameOverride is for testing only. If set to a non empty string,
+// it will override the virtual host name of authority (e.g. :authority header field) in requests.
+func NewClientTLSFromCert(cp *x509.CertPool, serverNameOverride string) TransportCredentials {
```

Comment:
That will be a massive API change, we don't want to do that unless necessary. In current state, this change has no influence for most users.

Another important reason is that we want to limit the use of `OverrideServerName()`. It should only be used in grpclb.
Calling `OverrideServerName()` after the creds is used in dialing can have unexpected behaviors, and we want to avoid the confusion.